lib/signal-list-view.coffee

```diff
@@ -58,18 +60,21 @@ class SignalListView extends SelectListView
                 kernel: kernel
             }
 
-        # add commands to switch to other kernels
-        @kernelManager.getAllKernelSpecsFor language, (kernelSpecs) =>
-            switchCommands = kernelSpecs.map (spec) ->
-                return {
-                    name: 'Switch to ' + spec.display_name
-                    value: 'switch-kernel'
-                    grammar: grammar
-                    language: language
-                    kernelSpec: spec
-                }
-
-            @setItems _.union basicCommands, switchCommands
+        if kernel instanceof WSKernel
```

Comment:
This is tricky to implement, so I left it for the future. The challenges are:
* Offering a selection of kernelspecs from the same gateway (there could be multiple gateways)
* Deciding whether to offer any local kernelspec options (if so, the UI must clearly distinguish them from any remote options)
* Cleanly shutting down the old kernel ("switch kernel" should kill the old kernel, by design)